Source/Core/Core/WiiUtils.cpp

```diff
@@ -74,4 +97,392 @@ bool InstallWAD(const std::string& wad_path)
+    if (!region.empty())
+      return region;
+  }
+
+  // Fall back to EUR.
+  return "EUR";
+}
+
+std::string SystemUpdater::GetDeviceId()
+{
+  u32 ios_device_id;
+  m_ios.GetES()->GetDeviceId(&ios_device_id);
+  return StringFromFormat("%" PRIu64, (u64(1) << 32) | ios_device_id);
+}
+
+constexpr const char* GET_SYSTEM_TITLES_REQUEST_PAYLOAD = R"END(\
+<?xml version="1.0" encoding="UTF-8"?>
+<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/"
+  xmlns:xsd="http://www.w3.org/2001/XMLSchema"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <soapenv:Body>
+    <GetSystemUpdateRequest xmlns="urn:nus.wsapi.broadon.com">
+      <Version>1.0</Version>
+      <MessageId>0</MessageId>
+      <DeviceId></DeviceId>
+      <RegionId></RegionId>
+    </GetSystemUpdateRequest>
+  </soapenv:Body>
+</soapenv:Envelope>
+)END";
+
+SystemUpdater::Response SystemUpdater::ParseTitlesResponse(const std::vector<u8>& response) const
+{
+  pugi::xml_document doc;
+  pugi::xml_parse_result result = doc.load_buffer(response.data(), response.size());
+  if (!result)
+    return {};
+
+  const pugi::xml_node node = doc.select_node("//*[local-name()='GetSystemUpdateResponse']").node();
+  if (!node)
+  {
+    ERROR_LOG(CORE, "ParseTitlesResponse: Could not find response node");
+    return {};
+  }
+
+  const int code = node.child("ErrorCode").text().as_int();
+  if (code != 0)
+  {
+    ERROR_LOG(CORE, "ParseTitlesResponse: Non-zero error code (%d)", code);
+    return {};
+  }
+
+  // libnup uses the uncached URL, not the cached one. However, that one is way, way too slow,
+  // so let's use the cached endpoint.
+  SystemUpdater::Response info;
+  info.content_prefix_url = node.child("ContentPrefixURL").text().as_string();
+  // Disable HTTPS because we can't use it without a device certificate.
+  info.content_prefix_url = ReplaceAll(info.content_prefix_url, "https://", "http://");
+  if (info.content_prefix_url.empty())
+  {
+    ERROR_LOG(CORE, "ParseTitlesResponse: Empty content prefix URL");
+    return {};
+  }
+
+  for (const pugi::xml_node& title_node : node.children("TitleVersion"))
+  {
+    const u64 title_id = std::stoull(title_node.child("TitleId").text().as_string(), nullptr, 16);
+    const u16 title_version = static_cast<u16>(title_node.child("Version").text().as_uint());
+    info.titles.push_back({title_id, title_version});
+  }
+  return info;
+}
+
+SystemUpdater::Response SystemUpdater::GetSystemTitles()
+{
+  // Construct the request by import the template first, then updating some fields.
+  pugi::xml_document doc;
+  pugi::xml_parse_result result = doc.load_string(GET_SYSTEM_TITLES_REQUEST_PAYLOAD);
+  _assert_(result);
+
+  // Nintendo does not really care about the device ID or verify that we *are* that device,
+  // as long as it is a valid Wii device ID.
+  const std::string device_id = GetDeviceId();
+  _assert_(doc.select_node("//DeviceId").node().text().set(device_id.c_str()));
```

Comment:
...but then again, this shouldn't work either (for the same reason pointed out above): `DeviceId` is an element in the namespace `urn:nus.wsapi.broadon.com` and shouldn't hit anything with that X-Path. Same below for the `RegionId`